hyperon_das/cache/iterators.py

```diff
@@ -434,7 +433,13 @@ def __next__(self):
             if self.buffer:
                 buffered_value, self.buffer = self.buffer, None
                 return buffered_value
-            link, targets = super().__next__()
+            link = super().__next__()
+            if isinstance(link, tuple):
+                link, targets = link
+            elif isinstance(link, dict):
+                targets = link.get('targets_document', [])
```

Comment:
I think it should be .pop(), otherwise it returns the 'targets_document' key in Atom to the user on line 449

Reply:
OK, I'll check this.